### PR TITLE
Make the asset path relative to the story

### PR DESCRIPTION
**Pull request by valendesigns**
Also fix updating of `canonical` link.

### CODE REVIEW

File: includes/sanitizers/class-amp-story-export-sanitizer.php [hunk 1]
```diff
@@ -108,19 +108,32 @@ public function sanitize() {
...
-					'href' => $this->args['canonical_url'],
-				]
-			);
-
-			$head = $this->dom->getElementsByTagName( 'head' )->item( 0 );
-			$head->appendChild( $rel_canonical );
+			$head          = $this->dom->getElementsByTagName( 'head' )->item( 0 );
+			$links         = [];
+			$link_elements = $head->getElementsByTagName( 'link' );
+			$rel_canonical = null;
```
Comment (valendesigns) [inline includes/sanitizers/class-amp-story-export-sanitizer.php]: I can remove this, it's not needed.

File: assets/src/stories-editor/plugins/export.js [hunk 2]
```diff
@@ -23,11 +23,30 @@ const handleExport = ( { postId, createErrorNotice, createSuccessNotice } ) => {
...
+	 * Show error notice.
+	 *
+	 * @param {?Error} error
+	 */
+	const showErrorNotice = ( error = null ) => {
+		removeNotice( progressNoticeId );
+		createErrorNotice( error ? error.message : errorMsg,
+			{
+				id: errorNoticeId,
+				type: 'snackbar',
```
Comment (valendesigns) [inline assets/src/stories-editor/plugins/export.js]: The error shouldn't be a snackbar.

File: assets/src/stories-editor/plugins/export.js [hunk 3]
```diff
@@ -56,21 +78,19 @@ const handleExport = ( { postId, createErrorNotice, createSuccessNotice } ) => {
...
 				// Handle the returned JSON error.
 				response.json()
 					.then( ( error ) => {
-						createErrorNotice( ( error.data && error.data.errorMessage ) ? error.data.errorMessage : errorMsg, {
-							id: 'amp-story-export__error-notice',
-						} );
-					} );
+						showErrorNotice( ( error.data && error.data.errorMessage ) ? new Error( error.data.errorMessage ) : null );
+					} )
+					.catch( showErrorNotice );
```
Comment (valendesigns) [inline assets/src/stories-editor/plugins/export.js]: Will writing it this way pass the error message to the function as a param automatically?

Comment (westonruter) [reply assets/src/stories-editor/plugins/export.js]: Yes. The callback is passed an `Error` object which then has a `message` property.

File: assets/src/stories-editor/plugins/export.js (hunk 2 above)
Comment (westonruter) [reply assets/src/stories-editor/plugins/export.js]: Fixed in cd68cb73

**Pull request merged**